discovery-provider/src/premium_content/premium_content_access_checker.py

```diff
@@ -0,0 +1,104 @@
+import json
+import logging
+from typing import Dict, Optional, Tuple, cast
+
+from src.models.tracks.track import Track
+from src.premium_content.helpers import does_user_have_nft_collection
+from src.premium_content.types import PremiumContentType
+from src.utils import db_session
+
+logger = logging.getLogger(__name__)
+
+
+class PremiumContentAccessChecker:
+    # check if content is premium
+    # - if not, then user has access
+    # - if so, check whether user fulfills the conditions
+    #
+    # Returns a tuple of (bool, bool) -> (track is premium, user has access)
+    #
+    # Note: premium content id for type should exist, but just in case it does not,
+    # we return true for user access so that (non-existent) track does not change
+    # existing flow of the function caller.
+    def check_access(
+        self,
+        user_id: int,
+        premium_content_id: int,
+        premium_content_type: PremiumContentType,
+    ) -> Tuple[bool, bool]:
```

Comment:
nit: can we have this method return a dictionary rather than two tuples? much more self documenting that way

I'd also opt for keyword args for methods 3+ params!